frame/support/procedural/src/pallet/parse/helper.rs

```diff
@@ -69,18 +70,29 @@ pub fn take_first_item_attr<Attr>(item: &mut impl MutItemAttrs) -> syn::Result<O
 }
 
 /// Take all the pallet attributes (e.g. attribute like `#[pallet..]`) and decode them to `Attr`
-pub fn take_item_attrs<Attr>(item: &mut impl MutItemAttrs) -> syn::Result<Vec<Attr>> where
+pub fn take_item_pallet_attrs<Attr>(item: &mut impl MutItemAttrs) -> syn::Result<Vec<Attr>> where
 	Attr: syn::parse::Parse,
 {
 	let mut pallet_attrs = Vec::new();
 
-	while let Some(attr) = take_first_item_attr(item)? {
+	while let Some(attr) = take_first_item_pallet_attr(item)? {
 		pallet_attrs.push(attr)
 	}
 
 	Ok(pallet_attrs)
 }
 
+/// Get all the cfg attributes (e.g. attribute like `#[cfg..]`) and decode them to `Attr`
+pub fn get_item_cfg_attrs(attrs: &Vec<syn::Attribute>) -> syn::Result<Vec<TokenStream>> {
```

Comment:
I don't think we need to loose the type, we could return `Vec<syn::Attribute>` here.

I think syn::Attribute implements ToToken so it can be used in quote in the expansion with no difficulty.